src/adapted/mod.rs

```diff
@@ -497,18 +509,22 @@ impl<'a> CustomTryFrom<(ReadDirection, raw::client_messages::ReadAllEventsComple
     fn try_from((dir, msg): (ReadDirection, raw::client_messages::ReadAllEventsCompleted<'a>)) -> Result<AdaptedMessage<'a>, ((ReadDirection, raw::client_messages::ReadAllEventsCompleted<'a>), Self::Err)> {
         use raw::client_messages::mod_ReadAllEventsCompleted::ReadAllResult;
 
-        let next_commit_position = LogPosition::from_i64_opt(msg.next_commit_position);
-        let next_prepare_position = LogPosition::from_i64_opt(msg.next_prepare_position);
+        // what is that ? is it ok to try parsing and if it fail just use None ?
+        let next_commit_position = LogPosition::try_from(msg.next_commit_position).ok();
+        let next_prepare_position = LogPosition::try_from(msg.next_prepare_position).ok();
```

Comment:
That is a good question. This might be a part of me being confused how to best handle these types. I think it might make more sense to return any errors on these two lines than to convert them to options.

I started writing these adapted forms perhaps too soon compared to how much I knew of ES back then.

In more broad scale I don't actually like the current `LogPosition` so much. The `LogPosition::First` and `LogPosition::Last` are only useful for the direction `client -> server`. In the responses `server -> client` the `-1` seems to mean invalid anyway (not last) if I have understood correctly.